frontend/src/pages/signup/hooks/useConfirmStatus.ts

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+
+export const useConfirmState = <T>(value: T) => {
+  const [lastConfirmedValue, setLastConfirmedValue] = useState<T | null>(null);
+  const [isCheckNeeded, setIsCheckNeeded] = useState(true);
+
+  const confirm = () => {
+    setLastConfirmedValue(value);
+    setIsCheckNeeded(true);
+  };
+
+  const shouldBlockSubmit = () => {
+    if (!isMatchedValue) {
+      setIsCheckNeeded(false);
+      return true;
+    }
+    return false;
+  };
+
+  const isMatchedValue = value === lastConfirmedValue;
```

Comment:
p4: 호이스팅이 되어서 isMatchedValue 를 사용할 수 있는것 같은데, 그래도 isMatchedValue가 위로 올라오는게 가독성이 더 좋을 것 같아!

```suggestion
  const isMatchedValue = value === lastConfirmedValue;

  const shouldBlockSubmit = () => {
    if (!isMatchedValue) {
      setIsCheckNeeded(false);
      return true;
    }
    return false;
  };
```

Reply:
이 훅에서 중요한 것들은 confrim과 shouldBlockSubmit 함수라고 생각해서 중요한 것부터 먼저 작성을 해서 파생상태가 밑으로 가게 구현됐어

난 보통 호이스팅 문제가 일어나지 않을때는 중요한 거부터 먼저 선언하는 것을 선호하는데 범태는 어때??

Reply:
나도 범태랑 같은 의견이였어! 
코드를 읽을 때 위에서부터 아래로 읽는데 isMatchedValue를 사용하고 있는데 없어서 어디서 가져온건지 모르겠었어!

위에 있으면 다음에 쓰겠구나 싶은데 어떻게 생각해? 🧐

Reply:
로건이 잘 얘기해줬다!!

Reply:
일단 우리끼리 혼동이될만한 거면 로건이랑 범태가 더 이해하기 쉬운쪽으로 해야하는게 맞는거 같아! 그러면 앞으로도 선언순으로 작성하게끔 신경쓸게 의견 고마워👍👍